File: Session3/973. K Closest Points to Origin.py
    def kClosest(self, points: List[List[int]], K: int) -> List[List[int]]:
        heap = []

        for (x, y) in points:
            d = -(x**2 + y**2)
            heapq.heappush(heap, (d, x, y))
            if len(heap) > K:
                heapq.heappop(heap)
        return [x[1:] for x in heap]

File: Session3/973. K Closest Points to Origin.java
package Session3;
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int []> pq=new PriorityQueue<>((a,b)->(-(a[0]*a[0]+a[1]*a[1]) + (b[0]*b[0]+b[1]*b[1]) ));
                                                            
            {
                pq.add(i);
                if(pq.size()>k)
                {
                    pq.poll();
                }
            }
            return pq.toArray(new int[0][]);
    }
}